bindings/python/src/logical_plan.rs

```diff
@@ -1,86 +1,191 @@
 use std::any::Any;
+use std::fmt::Debug;
 use std::sync::Arc;
 
+use arrow_util::pretty;
 use datafusion::arrow::datatypes::SchemaRef;
+use datafusion::arrow::pyarrow::ToPyArrow;
 use datafusion::datasource::TableProvider;
-use datafusion::error::Result as DatafusionResult;
+use datafusion::error::{DataFusionError, Result as DatafusionResult};
 use datafusion::execution::context::SessionState;
-use datafusion::logical_expr::{LogicalPlanBuilder, TableProviderFilterPushDown, TableType};
+use datafusion::execution::TaskContext;
+use datafusion::logical_expr::{TableProviderFilterPushDown, TableType};
+use datafusion::physical_plan::streaming::{PartitionStream, StreamingTableExec};
 use datafusion::physical_plan::ExecutionPlan;
 use datafusion::prelude::Expr;
+use futures::lock::Mutex;
+use futures::StreamExt;
+use glaredb::{RecordBatch, SendableRecordBatchStream};
+use pyo3::exceptions::PyRuntimeError;
 use pyo3::prelude::*;
-use sqlexec::{LogicalPlan, OperationInfo};
+use pyo3::types::PyTuple;
 
-use crate::connection::PyTrackedSession;
 use crate::error::PyGlareDbError;
-use crate::execution_result::PyExecutionResult;
 use crate::runtime::wait_for_future;
+use crate::util::pyprint;
 
 #[pyclass]
-#[derive(Clone, Debug)]
 pub struct PyLogicalPlan {
-    pub(super) lp: LogicalPlan,
-    pub(super) session: PyTrackedSession,
-    pub(super) op: OperationInfo,
+    schema: SchemaRef,
+    stream: SendableRecordBatchStream,
```

Comment:
I don't understand why there's a stream on this since we only get a stream after executing the plan.

Reply:
https://github.com/GlareDB/glaredb/pull/2832/files#diff-87aa6c7f95143a14b534fe4e79556ba4cfe9df4467c18b12f1d6d578e4882f50R130

Reply:
I agree, the `PyLogicalPlan` should be a thin wrapper around a `LogicalPlan`, not a `SendableRecordBatchStream`

Reply:
There are two reasons that I can see to have a public interface that has a concept of logical plans: 
- logical plans mirror the way that the db/datafusion operates internally, 
- maintaining the implementation details of the existing implementation.
Neither of these are, on their own particularly compelling to me, though perhaps I'm missing something. I think the naming is confusing and setting up a weird expectation, so I've cleaned that up. 

As I understand it, the public interface, exposed by the bindings, supports the following operations:
- "execute this query immediately in all cases, and provide access to the results."
- "parse this query, and if it's DDL operations, execute the operation immediately, and provide access to the results."
- "parse the query and plan it's execution, and provide access to the results, but do not execute the query until the results are accessed".
Where "provide access to the results" is typically:
- some sort of stream/iterator/cursor interface to a sequence of arrow record batches
- an object that can be referenced and used as a table in future queries.

Is there another user story or way of accessing the results?